2-й экран

Vekyork opened the pull request:
Сделал 2-й экран.
Прошу посмотреть.
Жду замечаний. Готов оперативно исправить недостатки.

Vekyork committed:
Сделал 2-й экран.
Прошу посмотреть.
Готов оперативно исправить недостатки.

Vekyork committed:
Merge branch 'Aristman:develop' into develop

Vekyork committed:
Сделал 2-й экран.
Прошу посмотреть.
Жду замечаний. Готов оперативно исправить недостатки.

zamyatinmn commented on app/build.gradle.kts:
```diff
@@ -64,6 +64,10 @@ dependencies {
+    implementation("com.google.android.material:material:1.5.0")
+    implementation("androidx.cardview:cardview:1.0.0")
+    implementation("androidx.constraintlayout:constraintlayout:2.1.3")
+    implementation("androidx.recyclerview:recyclerview:1.2.1")
```
Почти все эти зависимости уже подключены в проекте
А те, которых нет, необходимо описывать в классе Dependencies

Vekyork replied:
Удалил

zamyatinmn commented on app/src/main/res/values/dimens.xml:
```diff
@@ -20,4 +20,60 @@
+    <dimen name="property">18sp</dimen>
+    <dimen name="second_description">14sp</dimen>
+    <dimen name="_2">18sp bold</dimen>
+    <dimen name="V">18sp</dimen>
```
Что за имена? Через неделю даже автор не разберется что это

Vekyork replied:
V удалил
_1 и _2 это 1 и 2 в блоке перевода с разными вариантами перевода

Aristman replied:
Нейминг ниочем. Нужно все имена переменных давать осмысленными и понятными

zamyatinmn commented on app/src/main/res/values/icon_ruen.xml:
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Пустой файл, для чего он?

Vekyork replied:
Удалил

zamyatinmn commented on app/src/main/res/values-night/strings.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="poisk">poisk</string>
```
Необходимо избегать транслитерации

Vekyork replied:
Удалил poisk, везде search

zamyatinmn commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,239 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Этот макет нужно полностью переделать

Vekyork replied:
Полностью переделал макет

23err commented on app/build.gradle.kts:
```diff
@@ -64,6 +64,10 @@ dependencies {
+    implementation("com.google.android.material:material:1.5.0")
+    implementation("androidx.cardview:cardview:1.0.0")
+    implementation("androidx.constraintlayout:constraintlayout:2.1.3")
+    implementation("androidx.recyclerview:recyclerview:1.2.1")
```
implementation("com.google.android.material:material:1.5.0")
    implementation("androidx.cardview:cardview:1.0.0")
    implementation("androidx.constraintlayout:constraintlayout:2.1.3")
    implementation("androidx.recyclerview:recyclerview:1.2.1")
Все они уже установлены в проекте, можно удалить

23err commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,239 @@
+    android:layout_height="match_parent"
+    android:background="@color/primary_color"
+    tools:context=".MainActivity">
+    <RelativeLayout
```
Зачем в contraintlayout помещать relative? contraint это улучшенная версия Relative. Можно использовать только его. Если удобнее RelativeLayout, тогда лучше удалить constraint

23err commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,239 @@
+
+    <View
+            android:id="@+id/logo"
+            android:layout_width="141dp"
```
все размеры должны быть вынесены в ресурсы

23err commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,239 @@
+            android:layout_marginTop="168dp"
+            android:background="@drawable/search"
+            android:elevation="4dp"
+            tools:ignore="MissingConstraints" />
```
Зачем игнорировать предупреждение?

23err commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,239 @@
+
+        </FrameLayout>
+
+        <FrameLayout
```
Почему выбрал FrameLayout? у тебя много элементов в нем и элементы от размера экрана будут наезжать друг на друга

23err commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,239 @@
+                android:textAppearance="@style/second_description" />
+
+            <TextView
+                android:id="@+id/_2"
```
по id не понятно что за поле

23err commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,239 @@
+                android:layout_marginStart="267dp"
+                android:layout_marginTop="210dp"
+                android:gravity="center_horizontal|center_horizontal|center_vertical"
+                android:text="@string/V"
```
Что такое V?

23err commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,239 @@
+        android:layout_width="fill_parent"
+        android:layout_height="wrap_content"
+        app:menu="@menu/bottom_nav_menu"
+        tools:layout_editor_absoluteY="820dp" />
```
absolute значений не должно быть

Vekyork commented:
А что выбрать вместо него?08.04.2022, 21:53, "23err" ***@***.***>:
@23err commented on this pull request.



In app/src/main/res/layout/android_lar_2.xml:
> +            android:id="@+id/word_to_translate"
+            android:layout_width="178dp"
+            android:layout_height="50dp"
+            android:layout_alignStart="@+id/rus"
+            android:layout_alignParentTop="true"
+            android:layout_marginStart="62dp"
+            android:layout_marginTop="168dp"
+            android:background="#0fff0000"
+            android:gravity="center_horizontal|center_vertical"
+            ***@***.***/word_to_translate"
+            ***@***.***/word_to_translate"
+            tools:ignore="MissingConstraints">
+
+        </FrameLayout>
+
+        <FrameLayout

Почему выбрал FrameLayout? у тебя много элементов в нем и элементы от размера экрана будут наезжать друг на друга

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
-- 

Vekyork commented:
Это жирная 2 на экране перевода08.04.2022, 21:54, "23err" ***@***.***>:
@23err commented on this pull request.



In app/src/main/res/layout/android_lar_2.xml:
> +                android:gravity="center_horizontal|center_horizontal|center_vertical"
+                ***@***.***/property"
+                ***@***.***/property" />
+
+            <TextView
+                android:id="@+id/second_description"
+                android:layout_width="280dp"
+                android:layout_height="50dp"
+                android:layout_marginStart="20dp"
+                android:layout_marginTop="167dp"
+                android:gravity="center_horizontal|center_horizontal|center_vertical"
+                ***@***.***/second_description"
+                ***@***.***/second_description" />
+
+            <TextView
+                android:id="@+id/_2"

по id не понятно что за поле

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
-- 

Vekyork committed:
Переделал 2 экран. Остались мелочи.

Vekyork reviewed (commented):
Прошу посмотреть

23err commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,225 @@
+
+    <View
+        android:id="@+id/search"
+        android:layout_width="178dp"
```
Все значения надо вынести в ресурсы по всем view

Vekyork replied:
Исправил

23err commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,225 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout
```
Лучше все-таки использовать constraintLayout, relative уже устарел

Vekyork replied:
constraintLayout пробовал - всё плывет.

23err commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,225 @@
+
+    <TextView
+        android:id="@+id/rus"
+        android:layout_width="62dp"
```
ширина и высота редко задается конкретными параметрами. Т.к. на разных экранах может растягиваться и наоборот

Vekyork replied:
что тогда поставить вместо параметров ширины, высоты и привязки к Start и Top?

Vekyork replied:
Это ширина и высота конкретного блока. Как иначе. Я готов сразу переделать для многопользовательской версии. ПРОШУ уточнить - чем заменить ширину и высоту.

23err commented on app/src/main/res/layout/android_lar_2.xml:
```diff
@@ -0,0 +1,225 @@
+        android:layout_marginTop="80dp"
+        android:background="@drawable/logo" />
+
+    <View
```
За что эта view отвечает?

Vekyork replied:
Эта view отвечает за отображение логотипа

Vekyork committed:
Переделал 2 экран вновь. Все значения по всем view вынес в ресурсы. Убрал транслитерацию. удалил пустые и ненужные файлы. _1 и _2 это это ряды в блоке перевода для разных вариантов перевода. Убрал absolute значения.
Не отображается и не заполняется цветом background для длока перевода block_translate.

Vekyork committed:
Подправил немного

Vekyork committed:
Подправил немного

Vekyork committed:
ReNew
